src/os/ObjectStore.cc

```diff
@@ -185,3 +185,19 @@ int ObjectStore::queue_transactions(
   return queue_transactions(osr, tls, _onreadable, _oncommit,
 			    onreadable_sync, op);
 }
+
+uint64_t ObjectStore::estimate_objects_overhead( uint64_t num_objects, uint64_t num_whiteouts )
+{
+  if( blk_size == 0 ){
+    struct statfs stat;
+    int r = statfs(&stat);
+    if( r<0 ){
+       blk_size=4096; //let's use some assumed value to be able to estimate
+    }
+    else
+       blk_size=stat.f_bsize;
+  }
+  uint64_t res = num_objects * blk_size; //assumes that each object uses 1 additional block for attrs
+  res += (num_objects-num_whiteouts)*blk_size / 2; //assumes that each object except removed(i.e. empty) one uses ( in average ) additional 1/2 block due to FS allocation granularity.
+  return res;
+}
```

Comment:
Lots of whitespace issues here.  Please match the existing coding style

For both FileStore and BlueStore I think the object and whiteout overhead is the same.. I'm not sure we need to distinguish between the two.

And let's just make this pure virtual and define it in each one, since there are only 4 backends to implement.

BlueStore, KStore -> ~300 bytes
FileStore -> 2048 bytes
MemStore -> sizeof(Object) (doesn't really matter)